### ISSUE TITLE
Memory Leak?

### ISSUE THREAD

**Issue by michaelwan11** [openvla/openvla#39]
Hello, thanks for the great work!

I was wondering if there are any memory leaks? Maybe somewhere in the dataset/dataloader code? When I run the training code I get this plot of the process memory available.
![Screenshot 2024-07-22 at 12 20 54 AM](https://github.com/user-attachments/assets/7df5b3ed-d68a-4187-84e3-548774988db7)

Your help would be much appreciated!

**Comment by kpertsch** [comment]
Thanks for reaching out!
Yeah the RLDS data loader is notorious for filling up memory over time. Technically it saturates after some time, but in practice we restarted (resumed) training every couple days in case it crashed with an OOM error.
Fixing the underlying RLDS memory consumption is probably very hard (we weren't successful so far, even in the original Octo project where we built the data loader). 
If resuming is a big problem in your case, you can either reduce the shuffle buffer / number of data loading threads in the config or (more painful) convert things into a non-RLDS data format (like hdf5) which should bring down memory consumption a lot, but would require a full rewrite of the data loading infrastructure which we did not want to do.
